extensions/PhabBugz/lib/Feed.pm

```diff
@@ -358,6 +358,83 @@ sub _is_uplift_request_form_change {
   return $story_text =~ /\s+uplift request field/;
 }
 
+sub process_uplift_request_form_change {
+  # Process an uplift request form change for the passed revision object.
+  my ($revision, $changer) = @_;
+
+  my ($timestamp) = Bugzilla->dbh->selectrow_array("SELECT NOW()");
+
+  # Take no action if the form is empty.
+  # TODO test
+  if ($revision->uplift_form->is_empty) {
+    INFO("Uplift request form field cleared, ignoring.");
+    return;
+  }
+
+  my $revision_phid = $revision->phid;
+  INFO(
+    "Uplift request form submitted on $revision_phid, " .
+    "requesting `#release-managers` review."
+  );
+
+  # Get `#release-managers` review group.
+  my $release_managers_group = Bugzilla::Extension::PhabBugz::Project
+    ->new_from_query({name => 'release-managers'});
+
+  if (!$release_managers_group) {
+    WARN(
+      "Uplift request change detected but `#release-managers` was " .
+      "not found on Phabricator."
+    );
+    return;
+  }
+
+  my $release_managers_phid = $release_managers_group->phid;
+
+  # Request `#release-managers` review on each revision.
+  foreach my $stack_revision_phid (keys %{$revision->stack_graph_raw}) {
+    # Query Phabricator for each revision object related to the updated revision.
+    my $stack_revision = Bugzilla::Extension::PhabBugz::Revision->new_from_query(
+      {phids => [$stack_revision_phid]}
+    );
+
+    # Add `#release-managers!` review and set revision status.
+    $stack_revision->add_reviewer("blocking($release_managers_phid)");
+    $stack_revision->update();
+
+    INFO("Requested #release-managers review of $stack_revision_phid.");
+  }
+
+  # Comment the uplift form rendered on Bugzilla.
+  # TODO add rendered form to storage on Phabricator side.
+  # TODO test
+  my $comment_content = $revision->uplift_form->{"markdown"};
+  my $comment_params = {
+    'is_markdown' => 1,
+    'isprivate' => 0,
+  };
+  $revision->bug->add_comment($comment_content, $comment_params);
+
+  # If manual QE is required, set the Bugzilla flag.
+  # TODO test
+  if ($revision->uplift_form->{"Needs manual QE test"}) {
+    my $qe_flag = Bugzilla::FlagType->new({name => 'qe-verify'});
+    # TODO what to do if !$qe_flag ?
+    if ($qe_flag) {
+      my $new_flags = {
+        type_id  => $qe_flag->id,
+        status   => '?',
+        setter   => $changer->bugzilla_user->id,
```

Comment:
When this will execute, we have not yet ran set_phab_user() so the comment and flag setting might fail if the bug is private. Since the phabbugz daemon is running locally, no default user is set so we need to do it explicitly. 

We might be ok just moving set_phab_user earlier in process_revision_change() right before the uplift check.

Reply:
This is what I did and I can leave comments as that user in suite. :+1: